src/wallet/wallet.cpp

```diff
@@ -3372,10 +3372,8 @@ std::map<CTxDestination, CAmount> CWallet::GetAddressBalances() const
                     continue;
 
                 CAmount n = IsSpent(walletEntry.first, i) ? 0 : wtx.tx->vout[i].nValue;
-
-                if (!balances.count(addr))
-                    balances[addr] = 0;
-                balances[addr] += n;
+                auto ret = balances.emplace(addr, n);
+                if (!ret.second && n > 0) ret.first->second += n;
```

Comment:
I think it's nice to avoid a triple lookup here, but, I think the old code was much more readable.
Isn't there a clearer way to write this which keeps it clear that "n is always added to the balance"?

Wouldn't this simply work?:
```c++
                balances.emplace(addr, 0).second += n;
```

Reply:
Nice one, will use that if you don't mind.

Reply:
Why not `balances[addr] += n` ?

Reply:
Done, thanks!

Reply:
> Why not `balances[addr] += n` ?

Heh, I also thought about that, but then had doubts that integral types (without ctor) also get initialized. Seems I was wrong: "When invoking operator[] and the key is missing, the value is initialized using the expression mapped_type() which is the default constructor for class types, and zero initialization for integral types." https://stackoverflow.com/a/38549094

Reply:
🤦 

Done!